internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp

```diff
@@ -177,12 +181,15 @@ ChunkedSegmentSealedImpl::LoadScalarIndex(const LoadIndexInfo& info) {
     }
 
     scalar_indexings_[field_id] =
-        std::move(const_cast<LoadIndexInfo&>(info).index);
+        std::move(const_cast<LoadIndexInfo&>(info).cache_index);
+
+    auto accessor = SemiInlineGet(scalar_indexings_[field_id]->PinCells({0}));
+    auto scalar_filed_index = accessor->get_cell_of(0);
 
     set_bit(index_ready_bitset_, field_id, true);
     // release field column if the index contains raw data
     // only release non-primary field when in pk sorted mode
-    if (scalar_indexings_[field_id]->HasRawData() &&
+    if (scalar_filed_index->HasRawData() &&
```

Comment:
is it correct that for some indexes, HasRawData() is known only after the index is actually loaded? 

if so seems lazy loading index is not possible for now.

Reply:
maybe add a TODO to load index: to add meta such as expected_row_count and has_raw_data, so we don't need to load the entire index only to get such info.

Reply:
Static function can do it, let me optimize it